### PR TITLE
feat(Modal): Modal static function accept 'styles'

### PR DESCRIPTION
**Pull request by KotoriK**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #45580

### 💡 Background and solution
自5.10(#44934)，```bodyStyle```和```maskStyle```都被标记废弃。```ModalFuncProps```上对应的属性也被标为废弃，但是并没有对应地提供styles属性。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Modal static function accept 'styles'        |
| 🇨🇳 Chinese |  Modal静态方法接受styles属性         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 2d7d6a4</samp>

This pull request adds a new `styles` prop to the modal components and functions, which allows users to customize the modal appearance. It also includes a test case and some code refactoring.

### <samp>🤖 Generated by Copilot at 2d7d6a4</samp>

*  Add `styles` prop to modal components and functions to allow customizing the modal body and mask styles ([link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-d3734257bd9f28b52a331a7cc2fb5a3efd91c2fc71be29356d2fb171768ba9a3R188), [link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-ac9fa8e4784b40c5c2a26fe86dc50bbd39572b8ecf42b73611745ec817d883a1L11-R14), [link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-ac9fa8e4784b40c5c2a26fe86dc50bbd39572b8ecf42b73611745ec817d883a1L77-R79))
*  Use the `styles` prop value from the parent component or the default styles in the `Dialog` component ([link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-d3734257bd9f28b52a331a7cc2fb5a3efd91c2fc71be29356d2fb171768ba9a3L247-R248))
*  Remove the `styles` prop from the `ModalProps` interface as it is inherited from the `ModalCommonProps` interface ([link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-ac9fa8e4784b40c5c2a26fe86dc50bbd39572b8ecf42b73611745ec817d883a1L53))
*  Test that the `styles` prop works as expected in the `confirm.test.tsx` file ([link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-c211b371c72d68e91c25e16529bf8ded303122c95060910f58afdfc61a942a2bR666-R673))
*  Remove an extra space in the `footer` function call in the `ConfirmDialog.tsx` file ([link](https://github.com/ant-design/ant-design/pull/45558/files?diff=unified&w=0#diff-d3734257bd9f28b52a331a7cc2fb5a3efd91c2fc71be29356d2fb171768ba9a3L151-R153))

### CODE REVIEW

**Comment by KotoriK** [comment]
文档可能要填版本号就还没加

File: components/modal/interface.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ export type ModalFooterRender = (
   originNode: React.ReactNode,
   extra: { OkBtn: FC; CancelBtn: FC },
 ) => React.ReactNode;
-export interface ModalProps {
+interface ModalCommonProps{
```
Comment (afc163) [inline components/modal/interface.ts:11]:
```suggestion
interface ModalCommonProps {
```

**Review by afc163 — APPROVED** [review]

File: components/modal/ConfirmDialog.tsx [hunk 2]
```diff
@@ -148,9 +148,9 @@ export function ConfirmContent(
           <div className={`${confirmPrefixCls}-btns`}>
             {typeof footer === 'function'
               ? footer(footerOriginNode, {
-                  OkBtn,
-                  CancelBtn,
-                })
+                OkBtn,
+                CancelBtn,
+              })
```
Comment (afc163) [inline components/modal/ConfirmDialog.tsx:153]:
```suggestion
                  OkBtn,
                  CancelBtn,
                })
```

File: components/modal/ConfirmDialog.tsx [hunk 3]
```diff
@@ -244,7 +245,7 @@ const ConfirmDialog: React.FC<ConfirmDialogProps> = (props) => {
         mask={mask}
         maskClosable={maskClosable}
         style={style}
-        styles={{
+        styles={styles || {
           body: bodyStyle,
           mask: maskStyle,
         }}
```
Comment (afc163) [inline components/modal/ConfirmDialog.tsx:251]:
```suggestion
        styles={{
          body: bodyStyle,
          mask: maskStyle,
          ...styles,
        }}
```

Comment (afc163) [reply components/modal/ConfirmDialog.tsx:251]: ping

**Comment by afc163** [comment]
新特性发到 feature 分支吧。

**Pull request merged**